src/elrail.cpp

```diff
@@ -625,10 +625,12 @@ void SettingsDisableElrail(int32_t new_value)
 		}
 	}
 
-	/* Fix the total power and acceleration for trains */
-	for (Train *t : Train::Iterate()) {
-		/* power and acceleration is cached only for front engines */
-		if (t->IsFrontEngine()) {
+	for (const Company *c : Company::Iterate()) {
+		const GroupStatistics &stats = c->group_all[VEH_TRAIN];
+		/* Fix the total power and acceleration for trains */
+		for (const Vehicle *v : stats.vehicle_list) {
+			Train *t = Train::From(Vehicle::Get(v->index));
+			/* power and acceleration is cached only for front engines */
```

Comment:
This comment is no longer relevant, since it related to the original following line, that tests for `IsFrontEngine()`. But since your collection now only contains front engines, it no longer relates to this. The more appropriate location might be to move it outside the loop and note that the power/acceleration values are only cached for front engines and that is what this collection enumerates.